packages/shell-api/src/collection.ts

```diff
@@ -2131,39 +2177,42 @@ export default class Collection extends ShellApiWithMongoClass {
           const [host, numChunks] = await Promise.all([
             config
               .getCollection('shards')
-              .findOne({ _id: extShardStats.shard }),
+              .findOne({ _id: extractedShardStats.shard }),
             config.getCollection('chunks').countDocuments(countChunksQuery),
           ]);
           const shardStats = {
             shardId: shard,
             host: host !== null ? host.host : null,
-            size: extShardStats.storageStats.size,
-            count: extShardStats.storageStats.count,
+            size: extractedShardStats.storageStats.size,
+            count: extractedShardStats.storageStats.count,
             numChunks: numChunks,
-            avgObjSize: extShardStats.storageStats.avgObjSize,
+            avgObjSize: extractedShardStats.storageStats.avgObjSize,
           };
 
           const key = `Shard ${shardStats.shardId} at ${shardStats.host}`;
 
-          const estChunkData =
+          // In sharded timeseries collections, count is 0.
+          const shardStatsCount = shardStats.count ?? 0;
```

Comment:
A weird side-effect is that count (and consequently `docs` and `estimated docs per chunk`) will always be 0. This seems consistent with `collStats`

https://github.com/mongodb-js/mongosh/blob/a2fa6b458187ed32547e002d11d1492c7d7c016a/packages/shell-api/src/collection.ts#L1676-L1680

Is this what we want?

Reply:
Feels odd to say this, but this honestly sounds like a perfect use case for `NaN` – we don't know the number, and actually calculating it would be prohibitively expensive, so we might as well just forward this "we don't know" information to the caller?

Reply:
If I understand correctly, that'd mean actually returning intentionally returning `NaN` for `docs` and `estimated docs per chunk` in the result? Could this break something for potential users if they would otherwise expect some number? (i.e. maybe even Compass?)

Reply:
Compass doesn't use shell methods under the hood (currently)

I guess it could break scripts, yeah, but it still seems better than providing incorrect results, doesn't it?